src/main/java/org/springframework/integration/kafka/core/DefaultConnectionFactory.java

```diff
@@ -172,12 +103,17 @@ public void setFetchMetadataTimeout(int fetchMetadataTimeout) {
 	 */
 	@Override
 	public BrokerAddress getLeader(Partition partition) {
-		try {
-			lock.readLock().lock();
```

Comment:
Why `readLock` isn't needed anymore? Becuase need to upgrade to `writeLock` in `refreshLeaders` ? :smile: 
Anyway `getMetadataCache()` would be better for readability